secedgar/cik_lookup.py

```diff
@@ -1,27 +1,44 @@
 import warnings
 
-from secedgar.client.network_client import NetworkClient
-from secedgar.utils.cik_map import get_cik_map
-from secedgar.utils.exceptions import CIKError, EDGARQueryError
+import requests
 
+from secedgar.client import NetworkClient
+from secedgar.exceptions import CIKError, EDGARQueryError
 
-class _CIKValidator(object):
-    """Validates company tickers and/or company names based on CIK availability.
 
-    Used internally by the CIK class. Not intended for outside use.
+def get_cik_map(key="ticker"):
```

Comment:
Remove key parameter default. Also we should cache the requests.get call. My PR has code that only calls requests.get once.

Reply:
@reteps you are right that this should be cached so that we aren't making the same call twice. Only question is if you think that it is useful to have some sort of variant of `get_cik_map` available in the public API?

Reply:
@reteps I implemented caching, but kept it as a function. If there are multiple `CIKLookup` objects (for whatever reason), they can all use the cached `get_cik_map` result. Also, instead of returning a tuple, I return a dictionary with two keys, "ticker" and "title".

Let me know what you think of this change and any other feedback you have on the code.

Reply:
That's a good way of doing it

Reply:
left more changes